src/main/java/bgu/spl/net/impl/stomp/LibraryProtocol.java

```diff
@@ -13,59 +13,168 @@
     private Client _client;
     private int _connectionId;
     private Connections<Frame> _connections;
+    private Integer _messagesSent;
 
     @Override
     public void start(int connectionId, Connections<Frame> connections) {
         _connectionId = connectionId;
         _connections = connections;
+        _messagesSent = 0;
     }
 
     @Override
     public void process(Frame msg) {
         switch (msg.getCommand()) {
             case CONNECT:
-                _connections.send(_connectionId, handleConnect(msg));
+                handleConnect(msg);
                 break;
             case SEND:
                 handleSend(msg);
                 break;
-            // TODO: Add the rest of the handling. For all of the commands
+            case SUBSCRIBE:
+                handleSubscribe(msg);
+                break;
+            case UNSUBSCRIBE:
+                handleUnsubscribe(msg);
+                break;
+            case DISCONNECT:
+                handleDisconnect(msg);
+                break;
         }
     }
 
+    private void handleDisconnect(Frame msg) {
+        sendReceipt(msg);
+        _connections.disconnect(_connectionId);
+        _client.connected = false;
+        _shouldTerminate = true;
+    }
+
     /**
-     *  Adds a receipt id needed according to {@code msg}'s headers
+     * This function checks for the existence of the header {@code header}, if it doesn't exist the function sends the
+     * proper error Frame
+     *
+     * @param msg    The Frame to check for the header
+     * @param header The header to be checked for existence
+     * @return Does the header exist
+     */
+    private boolean checkForHeader(Frame msg, String header, String info) {
+        if (msg.getHeader(header) == null) {
+            Frame f = createErrorFrame("MalFormed Frame - missing " + header + " header");
+            if (info != null) addErrorBody(msg, f, info);
+            _connections.send(_connectionId,
+                    f);
```

Comment:
redundant newline

Reply:
True, weird why it was added in the first place